linguistlab.py

```diff
@@ -1,20 +1,29 @@
 import csv
-import sys
+import pathlib
 
+from tkinter import filedialog
 from typing import Dict, List, Any, Optional
 
 
-def read_glossary(csv_file: Any) -> Dict[str, List[str]]:
+# TODO: Handle wrongly formatted glossary/csv files
+def open_glossary() -> "Glossary":
+    path = pathlib.Path(filedialog.askopenfilename())
+
+    if path.suffix != ".csv":
+        raise TypeError("LinguistLab currently only supports glossaries in .csv format.")
+
+    glossary_name = path.name.split(".")[0].title()
```

Comment:
```
>>> from pathlib import Path
>>> Path("/foo/bar/asdf.txt").stem
'asdf'
```

So you could do:

```suggestion
    glossary_name = path.stem.title()
```

This also handles the corner case where your path starts with a dot, which is typical for hidden files: `.secretglossary.csv`.

Another thing to note is that `.title()` doesn't handle `'s` very well:

```
>>> Path("/home/akuli/akuli's glossary.csv").stem.title()
"Akuli'S Glossary"
```

I personally would just display `path.name` or `path.stem` to the user as is.